src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -62,9 +62,17 @@ public function getEnvPlaceholders()
 
     /**
      * Merges the env placeholders of another EnvPlaceholderParameterBag.
+     *
+     * @param EnvPlaceholderParameterBag $bag
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
+
+        foreach ($newPlaceholders as $key => $newEntries) {
```

Comment:
`reset` has two drawbacks: it works by reference, which I prefer to avoid, and it changes the internal state of the array, which is a not wanted side effect. Even if these could be of no importance in this case, I consider using it a worse practice...